### PR TITLE
In RC doc, add deployment & RS as alternatives, and link to kubectl rolling-update tutorial

### PR DESCRIPTION
**Pull request by janetkuo**
@kubernetes/docs

### CODE REVIEW

File: docs/user-guide/replication-controller/index.md [hunk 1]
```diff
@@ -226,18 +226,30 @@ object](/docs/api-reference/v1/definitions/#_v1_replicationcontroller).
...
 
+### ReplicaSet
+
+[`ReplicaSet`](/docs/user-guide/replicasets/) is the next-generation Replication Controller that supports the new [set-based label selector](/docs/user-guide/labels/#set-based-requirement).
+It’s mainly used by [`Deployment`](/docs/user-guide/deployments/) as a mechanism to orchestrate pod creation, deletion and updates.
+Note that we recommend using Deployments instead of directly using Replica Sets, unless you require custom update orchestration or don’t require updates at all.
+
+### Deployment (Recommended)
+
+[`Deployment`](/docs/user-guide/deployments/) is a higher level API object that updates Pods (and its underlying Replica Sets)
```
Comment (bgrant0607) [inline docs/user-guide/replication-controller/index.md]: higher-level

English is right-associative. :-)

"updates its underlying Replica Sets and their Pods"

File: docs/user-guide/replication-controller/index.md [hunk 2]
```diff
@@ -226,18 +226,30 @@ object](/docs/api-reference/v1/definitions/#_v1_replicationcontroller).
...
+
+[`ReplicaSet`](/docs/user-guide/replicasets/) is the next-generation Replication Controller that supports the new [set-based label selector](/docs/user-guide/labels/#set-based-requirement).
+It’s mainly used by [`Deployment`](/docs/user-guide/deployments/) as a mechanism to orchestrate pod creation, deletion and updates.
+Note that we recommend using Deployments instead of directly using Replica Sets, unless you require custom update orchestration or don’t require updates at all.
+
+### Deployment (Recommended)
+
+[`Deployment`](/docs/user-guide/deployments/) is a higher level API object that updates Pods (and its underlying Replica Sets) 
+in a similar fashion as `kubectl rolling-update`. Deployments are recommended if you want this rolling update functionality, 
+because unlike `kubectl rolling-update`, they are declarative, server side, and have additional features.
```
Comment (bgrant0607) [inline docs/user-guide/replication-controller/index.md]: server-side

File: docs/user-guide/replication-controller/index.md (hunk 1 above)
Comment (janetkuo) [inline docs/user-guide/replication-controller/index.md]: Fixed

File: docs/user-guide/replication-controller/index.md (hunk 2 above)
Comment (janetkuo) [inline docs/user-guide/replication-controller/index.md]: Fixed

**Comment by janetkuo** [comment]
@bgrant0607 all comments addressed; PTAL

**Comment by bgrant0607** [comment]
LGTM, thanks.

**Pull request merged**